### PR TITLE
Fix matrix multiplication

### PR DESCRIPTION
**Pull request by toor1245**

### CODE REVIEW

File: tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs [hunk 1]
```diff
@@ -87,6 +88,64 @@ public void MatrixMultiplySimdDoubleTest_MultiplyTwoMatrix_AssertMustBeEqual()
...
+                    {
+                        var b = pointer2 + k * n;
+                        var a = span1[i * K + k];
+                        for (var j = 0; j < n; j++) c[j] += a * b[j];
+                    }
+                }
+
+                return matrix;
+            }
+        }
```
Comment (Dilorfin) [inline tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs:148]: What a f is here we have?)
Are you sure that we should use "unsafe code" is Unit Tests?
Obviously, I don't think so! As for me code in unit tests must be as simple&stupid as it only can be

File: tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs [hunk 2]
```diff
@@ -87,6 +88,64 @@ public void MatrixMultiplySimdDoubleTest_MultiplyTwoMatrix_AssertMustBeEqual()
...
+            var matrixB = new double[size, size];
+            for (int i = 0; i < matrixB.GetLength(0); i++)
+            {
+                for (int j = 0; j < matrixB.GetLength(1); j++)
+                {
+                    matrixB[i, j] = 1;
+                }
+            }
+            
+            var expected = MulMatrixTest(matrixB, matrixB);
```
Comment (Dilorfin) [inline tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs:111]: Multiplication result for square matrixes that consist only of `1` will be just matrix `size`x`size` that consist of `size`

**Review by Dilorfin — CHANGES_REQUESTED** [review]

File: tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs (hunk 1 above)
Comment (toor1245) [reply tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs:148]: it is ctrl C + Ctrl V for testing

Comment (Dilorfin) [reply tests/MatrixDotNet.UnitTesting/MatrixTests/BasicOperations.cs:148]: but in the end it doesn't even matter

**Review by Dilorfin — APPROVED** [review]

**Pull request merged**